myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/JHSTaskFactoryImpl.java

```diff
@@ -0,0 +1,203 @@
+        if (returnedPorts != null && !returnedPorts.isEmpty()) {
+          Value.Ranges.Builder valueRanger = Value.Ranges.newBuilder();
+          for (Long port : returnedPorts) {
+            valueRanger.addRange(Value.Range.newBuilder()
+                    .setBegin(port)
+                    .setEnd(port));
+          }
+          
+          taskBuilder.addResources(Resource.newBuilder().setName("ports")
+              .setType(Value.Type.RANGES)
+              .setRanges(valueRanger.build()));
+        }
+        taskBuilder.setCommand(commandInfo);
+        return taskBuilder.build();
+      }
+      
+      /**
+       * Helper method to reserve ports
+       * @param offer
+       * @param requestedPorts
+       * @return
+       */
+      private List<Long> getAvailablePorts(Offer offer, List<Long> requestedPorts) {
+        if (requestedPorts == null || requestedPorts.isEmpty()) {
+          return null;
+        }
+        final List<Long> returnedPorts = new ArrayList<>();
+        boolean [] retValue = new boolean[requestedPorts.size()];
+        for (Resource resource : offer.getResourcesList()){
+          if (resource.getName().equals("ports")){
+            Iterator<Value.Range> itr = resource.getRanges().getRangeList().iterator();
+            int l = 0;
+            while (itr.hasNext()) {
+              Value.Range range = itr.next();
+              if (range.getBegin() <= range.getEnd()) {
+                for (int k = 0; k < requestedPorts.size(); k++) {
+                  final Long port = requestedPorts.get(k);
+                  if (port <= range.getEnd() && port >= range.getBegin()) {
+                    retValue[k] = true;
+                    l++;
+                  }
+                  if (l >= requestedPorts.size()) {
+                    return requestedPorts;
+                  }
+                }
+                // fill up some ports in case we won't find ones we need
+                long i = range.getBegin();
+                while (i <= range.getEnd() && returnedPorts.size() < requestedPorts.size()) {
+                  returnedPorts.add(i);
+                  i++;
+                }
+              }
+            }
+          }
+        }
+        // in case we managed to get some required ports
+        // set to random only ones we could not set to requested
+        for (int j = 0; j < retValue.length; j++) {
+          if (retValue[j]) {
+            returnedPorts.set(j, requestedPorts.get(j));
+          } else {
+            // TODO (yfeldman) need to make this more visible
+            LOGGER.info("Following port: {} is not available will use {} port instead", 
+                requestedPorts.get(j), returnedPorts.get(j));
+          }
+        }
+        return returnedPorts;
+      }
+      @Override
+      public ExecutorInfo getExecutorInfoForSlave(SlaveID slave) {
+        // nothing to implement here, since we are using default slave executor
+        return null;
+      }      
+    }
```

Comment:
Doesn't seem to support remote distribution so I can't test.  I'll make comments that I see from the code for now.  I think #118 will make make this easier to add.